packages/webdriverio/src/find-strategy.js

```diff
@@ -0,0 +1,183 @@
+import { W3C_SELECTOR_STRATEGIES } from './constants'
+
+const DEFAULT_STRATEGY = 'css selector'
+const DIRECT_SELECTOR_REGEXP = /^(id|css selector|xpath|link text|partial link text|name|tag name|class name|-android uiautomator|-ios uiautomation|-ios predicate string|-ios class chain|accessibility id):(.+)/
+const XPATH_SELECTORS_START = [
+    '/', '(', '\'../\'', './', '*/'
+]
+const NAME_MOBILE_SELECTORS_START = [
+    'uia', 'xcuielementtype', 'android.widget'
+]
+const XPATH_SELECTOR_REGEXP = [
+    // HTML tag
+    /^([a-z0-9|-]*)/,
+    // optional . or # + class or id
+    /(?:(\.|#)(-?[_a-zA-Z]+[_a-zA-Z0-9-]*))?/,
+    // optional [attribute-name="attribute-selector"]
+    /(?:\[(-?[_a-zA-Z]+[_a-zA-Z0-9-]*)(?:=(?:"|')([a-zA-z0-9\-_. ]+)(?:"|'))?\])?/,
+    // *=query or =query
+    /(\*)?=(.+)$/,
+]
+
+const defineStrategy = function (selector) {
+    /**
+     * check if user has specified locator strategy directly
+     */
+    if (selector.match(DIRECT_SELECTOR_REGEXP)) {
+        return 'directly'
+    }
+    // use xPath strategy if selector starts with //
+    if (XPATH_SELECTORS_START.some(option => selector.indexOf(option) === 0)) {
+        return 'xpath'
+    }
+    // use link text strategy if selector starts with =
+    if (selector.indexOf('=') === 0) {
+        return 'link text'
+    }
+    // use partial link text strategy if selector starts with *=
+    if (selector.indexOf('*=') === 0) {
+        return 'partial link text'
+    }
+    // recursive element search using the UiAutomator library (Android only)
+    if (selector.indexOf('android=') === 0) {
+        return '-android uiautomator'
+    }
+    // recursive element search using the UIAutomation library (iOS-only)
+    if (selector.indexOf('ios=') === 0) {
+        return '-ios uiautomation'
+    }
+    // recursive element search using accessibility id
+    if (selector.indexOf('~') === 0) {
```

Comment:
let's use `startsWith` instead of `indexOf`, much more expressive imo

Reply:
totally agre